platform/darwin/src/MGLSource.mm

```diff
@@ -33,13 +36,18 @@ - (instancetype)initWithIdentifier:(NSString *)identifier
 - (instancetype)initWithRawSource:(mbgl::style::Source *)rawSource mapView:(MGLMapView *)mapView {
     NSString *identifier = @(rawSource->getID().c_str());
     if (self = [self initWithIdentifier:identifier]) {
-        _rawSource = rawSource;
-        _rawSource->peer = SourceWrapper { self };
+        _weakSource = rawSource->makeWeakPtr();
+        rawSource->peer = SourceWrapper { self };
         _mapView = mapView;
     }
     return self;
 }
 
+- (mbgl::style::Source *)rawSource
```

Comment:
Need to look at any nullability concerns here - 

Though I think it's only the `MGLRasterTileSource` `rawSource` property that might need to be tweaked as nullable - though this is still a private header, so not a huge pain:

https://github.com/mapbox/mapbox-gl-native/blob/d41f6c9f74580a8825da55b9ca63587b0ff1195f/platform/darwin/src/MGLRasterTileSource_Private.h#L11-L15